### PR TITLE
Communicator trash folder archive task

### LINKED ISSUE #5292: Viestimen roskakorin automaattinen tyhjennys
Roskakorissa oleva viesti voisi poistua automaattisesti, kun se on ollut roskakorissa 30 vuorokautta.

### PR DESCRIPTION
**Pull request by aviljakaine**
Maybe add a message to UI about the messages being cleaned from trash after 30 days.

Closes #5292

### CODE REVIEW

**Review by aviljakaine — COMMENTED** [review]
Major problem with this PR: doesn't refresh search index for removed messages.

File: updates/1.141/230929-communicator-trash-timer.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pkukkon) [inline updates/1.141/230929-communicator-trash-timer.xml:1]: Seems OK but version 1.141 has already been claimed.

**Review by saaralepisto — APPROVED** [review]
backend seems ok

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/messages.tsx [hunk 2]
```diff
@@ -409,6 +393,13 @@ class CommunicatorMessages extends BodyScrollLoader<
 
     return (
       <BodyScrollKeeper hidden={!!this.props.currentThread}>
+        {this.props.messages.location === "trash" ? (
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/messages.tsx:396]: Tän vois tehdä lyhyemmin ja ilman else-käsittelyä ihan vaan :

`{this.props.messages.location === "trash" && ...}`

Ei merkittävä asia sinänsä.

**Review by avukkonen — CHANGES_REQUESTED** [review]

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/messages.tsx (hunk 2 above)
Comment (aviljakaine) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/messages.tsx]: Muutettu

File: updates/1.141/230929-communicator-trash-timer.xml (hunk 1 above)
Comment (aviljakaine) [reply updates/1.141/230929-communicator-trash-timer.xml]: Should be fine now

File: updates/1.140/230929-communicator-trash-timer.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (aviljakaine) [reply updates/1.140/230929-communicator-trash-timer.xml]: This was added

**Review by avukkonen — APPROVED** [review]

**Review by pkukkon — APPROVED** [review]
Backend OK

**Pull request merged**